ui/src/components/UploadStory.tsx

```diff
@@ -0,0 +1,855 @@
+import useSWR from "swr";
+
+import { StoryDrawer } from "../components";
+import citiesList from "../data/cities.json";
+import { colors } from "../styles/colors";
+import { device } from "../styles/device";
+import { UploadLabelsText, UploadStoriesHeading } from "../styles/typography";
+import { Author, Story } from "../types";
+
+const StyledGrid = styled(Grid)`
+  background-color: ${colors.primaryLight6};
+  @media ${device.laptop} {
+    justify: left;
+    width: 100vw;
+  }
+`;
+
+const StyledSelect = styled(Select)`
+  width: 30vw;
+  && {
+    font-family: Poppins;
+    align-items: center;
+    display: inline-block;
+  }
+`;
+
+const StyledMenuItem = styled(MenuItem)`
+  && {
+    font-family: Poppins;
+  }
+`;
+
+const StyledTextField = styled(TextField)`
+  width: 30vw;
+
+  && {
+    margin-top: 12px;
+    .MuiInputLabel-formControl {
+      font-family: Poppins !important;
+    }
+  }
+`;
+
+const ImageContainer = styled.div`
+  width: 30vw;
+`;
+
+const StyledDropzoneArea = styled(DropzoneArea)`
+  && {
+    margin-top: 12px;
+  }
+`;
+
+const StyledAutocomplete = styled(Autocomplete)`
+  && {
+    font-family: Poppins;
+    margin-top: 12px;
+    width: 30vw;
+    .MuiInputLabel-root {
+      font-family: Poppins !important;
+    }
+    .MuiInputBase-input {
+      font-family: Poppins !important;
+    }
+  }
+`;
+
+const StyledButton = styled(Button)`
+  text-transform: none;
+  margin: 5px;
+`;
+
+const AddCountryPaper = styled(Paper)`
+  width: 30vw;
+  font-family: Poppins !important;
+`;
+
+const AddCountryDiv = styled.div`
+  margin-left: 18px;
+  margin-top: 5px;
+`;
+
+const AddCountryButton = styled(Button)`
+  margin-right: 5px;
+  width: 2vw;
+  float: right;
+  height: 3.75vh;
+
+  .MuiButton-label {
+    color: ${colors.primaryDark1};
+    font-family: Poppins !important;
+  }
+`;
+const StyledBackgroundColor = styled.div`
+  background-color: ${colors.white};
+  width: 40vw;
+  padding: 0px 0px 24px 24px;
+  margin-bottom: 24px;
+`;
+
+const StyledLinearProgress = styled(LinearProgress)`
+  margin-top: 24px;
+`;
+
+const StyledImage = styled.img`
+  width: 30vw;
+`;
+
+const LISTBOX_PADDING = 8; // px
+
+function renderRow(props: ListChildComponentProps) {
+  const { data, index, style } = props;
+  return React.cloneElement(data[index], {
+    style: {
+      ...style,
+      top: (style.top as number) + LISTBOX_PADDING,
+    },
+  });
+}
+
+const OuterElementContext = React.createContext({});
+
+const OuterElementType = React.forwardRef<HTMLDivElement>((props, ref) => {
+  const outerProps = React.useContext(OuterElementContext);
+  return <div ref={ref} {...props} {...outerProps} />;
+});
+
+function useResetCache(data: any) {
+  const ref = React.useRef<VariableSizeList>(null);
+  React.useEffect(() => {
+    if (ref.current != null) {
+      ref.current.resetAfterIndex(0, true);
+    }
+  }, [data]);
+  return ref;
+}
+
+// Adapter for react-window
+const ListboxComponent = React.forwardRef<HTMLDivElement>(
```

Comment:
Can we separate this `ListboxComponent` into another file?

Reply:
Will do, but tbh I need to ask you and @AbhiPrasad about this because I'm not sure if it's actually doing anything lol